Move operator-sdk to v1.32.0

ramperher opened the pull request:
- Update operator-sdk to v1.32.0
- Update operator_sdk.util collection to latest version, v0.5.0 (https://galaxy.ansible.com/ui/repo/published/operator_sdk/util/)
- Update CHANGELOG.md, including versions not reported in the corresponding files, following https://github.com/openshift-kni/example-cnf/pull/27 to avoid merge conflicts afterwards

ramperher commented:
> * FAILURE https://www.distributed-ci.io/jobs/11550c7b-9fc4-42d8-9ad2-4578e50c0a07/jobStates
>   from [Move operator-sdk to v1.32.0 #26](https://github.com/openshift-kni/example-cnf/pull/26)

I tried to run the teardown to remove example-cnf resources in the cluster where I'm going to test this change and it was failing because cnf_namespace variable is not defined, fixing it here: https://github.com/dci-labs/example-cnf-config/pull/88

ramperher commented:
> * FAILURE https://www.distributed-ci.io/jobs/a919111d-db59-462b-8fc2-d074fd1f756e/jobStates
>   from [Move operator-sdk to v1.32.0 #26](https://github.com/openshift-kni/example-cnf/pull/26)

Last debug job was testing the example-cnf installation and failed in the MCP check, not related to this change. The teardown passed, in fact. Change is ready for review.

capolrik commented:
Arg, there is something wrong in the build in the CI. When I look at the "operators.operatorframework.io/builder" [here](https://www.distributed-ci.io/jobs/a919111d-db59-462b-8fc2-d074fd1f756e/jobStates?sort=date&task=00450fd4-de62-4411-a2c3-71730db2f71a), it says "operators.operatorframework.io/builder": "operator-sdk-v1.31.0" and 1.32.0.

tonyskapunk commented:
Is still using 1.31.0: https://github.com/openshift-kni/example-cnf/actions/runs/7057725984/job/19211911169?pr=26#step:4:162 it is overridden from the main Makefile, you want to change that one too or maybe not at all :smiley_cat: 

https://github.com/openshift-kni/example-cnf/blob/dccd3973330a0f056053301fb49ab0a60cc9e005/Makefile#L5

tonyskapunk commented:
This change introduced that override: https://github.com/openshift-kni/example-cnf/pull/18 you might want to talk to @tkrishtop as the build is always overriding the default version.

ramperher commented:
> Is still using 1.31.0: https://github.com/openshift-kni/example-cnf/actions/runs/7057725984/job/19211911169?pr=26#step:4:162 it is overridden from the main Makefile, you want to change that one too or maybe not at all 😺
> 
> https://github.com/openshift-kni/example-cnf/blob/dccd3973330a0f056053301fb49ab0a60cc9e005/Makefile#L5

Oh, that's true, I forgot to push that :sweat_smile: now updated

ramperher commented:
I've updated the change to also update operator_sdk.util collection, since it's not using the latest version right now. And also, I've updated the CHANGELOG properly, but since latest version was not reported, I had to create https://github.com/openshift-kni/example-cnf/pull/27 to solve that.

After this, I'll wait for the Github action before running a test

ramperher commented:
As I've said in CILAB-1282, it's quite weird that, for all the example-cnf tests I am launching (it's true I'm not launching the cluster from scratch, but I'm reusing the cluster), I'm always having a Ready,SchedulingDisabled issue when finishing preflight testing, example here: https://www.distributed-ci.io/jobs/e5aedfcc-ee15-4328-883e-dd2826b04508/jobStates?sort=date&task=d4c0b188-d904-4647-9fd3-2ccd3bfa07c2

And even though I uncordon the affected node, the MCP does not progress to ready state:

```
[kni@provisioner.cluster2.dfwt5g.lab ~]$ oc get nodes                                                   
NAME       STATUS                     ROLES                  AGE   VERSION                              
master-0   Ready                      control-plane,master   13h   v1.27.6+d548052                      
master-1   Ready                      control-plane,master   13h   v1.27.6+d548052                      
master-2   Ready                      control-plane,master   13h   v1.27.6+d548052                      
worker-0   Ready,SchedulingDisabled   worker                 12h   v1.27.6+d548052                      
worker-1   Ready                      worker                 12h   v1.27.6+d548052                      
worker-2   Ready                      worker                 12h   v1.27.6+d548052                      
worker-3   Ready                      worker                 12h   v1.27.6+d548052                      
[kni@provisioner.cluster2.dfwt5g.lab ~]$ oc adm uncordon worker-0                                       
node/worker-0 uncordoned                       
[kni@provisioner.cluster2.dfwt5g.lab ~]$ oc get nodes
NAME       STATUS   ROLES                  AGE   VERSION
master-0   Ready    control-plane,master   13h   v1.27.6+d548052
master-1   Ready    control-plane,master   13h   v1.27.6+d548052
master-2   Ready    control-plane,master   13h   v1.27.6+d548052
worker-0   Ready    worker                 12h   v1.27.6+d548052
worker-1   Ready    worker                 12h   v1.27.6+d548052
worker-2   Ready    worker                 12h   v1.27.6+d548052
worker-3   Ready    worker                 12h   v1.27.6+d548052
(...)
[kni@provisioner.cluster2.dfwt5g.lab ~]$ oc get mcp
NAME     CONFIG                                             UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
master   rendered-master-c1603bbb226063f71d2e990e3da44351   True      False      False      3              3                   3                     0                      13h
worker   rendered-worker-b07d9874b175c6df262fc303a60ad28c   False     True       False      4              1                   1                     0                      13h
```

The job I've launched will probably fail, but all the installation and tests have passed correctly. So I don't know if we should go ahead with this, or if maybe I'm introducing something that is making the nodes to be in SchedulingDisabled (but I don't know in that case what's going on).

ramperher commented:
I'm going to run a job from scratch just in case

ramperher commented:
check dallas ocp-4.14-vanilla example-cnf

tonyskapunk commented:
> > Is still using 1.31.0: https://github.com/openshift-kni/example-cnf/actions/runs/7057725984/job/19211911169?pr=26#step:4:162 it is overridden from the main Makefile, you want to change that one too or maybe not at all 😺
> > https://github.com/openshift-kni/example-cnf/blob/dccd3973330a0f056053301fb49ab0a60cc9e005/Makefile#L5
> 
> Oh, that's true, I forgot to push that 😅 now updated

I guess I wasn't clear enough, my bad. This override was introduced in #18 my question is, do we still need to override? or can we remove it and use the version each operator has?

ramperher commented:
> > > Is still using 1.31.0: https://github.com/openshift-kni/example-cnf/actions/runs/7057725984/job/19211911169?pr=26#step:4:162 it is overridden from the main Makefile, you want to change that one too or maybe not at all 😺
> > > https://github.com/openshift-kni/example-cnf/blob/dccd3973330a0f056053301fb49ab0a60cc9e005/Makefile#L5
> > 
> > 
> > Oh, that's true, I forgot to push that 😅 now updated
> 
> I guess I wasn't clear enough, my bad. This override was introduced in #18 my question is, do we still need to override? or can we remove it and use the version each operator has?

Oh, got it. Honestly, I'd bet for removing it from the main Makefile, but since @tkrishtop did the change and she'll not return till next week, maybe we can merge this change and target this other update in a separate change if needed, what do you think, @tonyskapunk ?

By the way, the job is working, see the latest clean job launched yesterday. This would be ready for merging.

tonyskapunk commented:
> target this other update in a separate change if needed,

Agreed

tonyskapunk reviewed (approved):
:+1:
